### PR TITLE
Import code from BitBucket

### PR DESCRIPTION
**Pull request by jirutka**
This PR basically includes changes from ce69a1c84c16c2dd6084171250190461d4996638 and 605b5e558ecd28458c5be82bee4bfb767ef6bb9a divided into smaller commits, plus few fixes.

Do not merge it yet!

### CODE REVIEW

File: Dockerfile.armv7l [hunk 1]
```diff
@@ -7,6 +7,6 @@ RUN cd /tmp/bc-hub;./build.sh
 ENV BLYNK_TOKEN=???, BLYNK_MIN_PUBLISH_INTERVAL=10
 
 # Caddy web server, Mosquitto MQTT broker
-EXPOSE 80 1883
+EXPOSE 80 443 1883
```
Comment (muhlpachr) [inline Dockerfile.armv7l]: @jirutka Why expose 443 in container configuration where nothing listens on TCP port 443 ?

Comment (jirutka) [inline Dockerfile.armv7l]: Caddy still listens on 443, for backward compatibility (see 6b64a482bd68d9ce1e32107aecb01184e9f16532).

File: build.sh [hunk 2]
```diff
@@ -65,11 +59,8 @@ mkdir -p "$WWW_DIR"/auth/dashboard
 cp dist/* "$WWW_DIR"/auth/dashboard/
 install "$SRC_DIR"/etc/$DASHBOARD_CONFIG "$WWW_DIR"/auth/dashboard/config.js
 
-echo '==> Installing quick start documentation...'
-cp "$SRC_DIR"/README.md "$WWW_DIR"/
-#cp -r "$SRC_DIR"/doc "$WWW_DIR"/
```
Comment (muhlpachr) [inline build.sh]: @jirutka We need to add bc-doc submodule and add this copy

File: Dockerfile.armv7l (hunk 1 above)
Comment (muhlpachr) [inline Dockerfile.armv7l]: ok, we still need to remove it before release, so why not now ? ;-)

Comment (jirutka) [inline Dockerfile.armv7l]: I’m actually not sure if we can/should remove it (the note about removing is outdated). From my PR on BitBucket:

> When you open `http://<address>/`, browser may redirect you to `https://<address>/auth/dashboard` that doesn’t work anymore. The problem is that there used to be a _permanent_ redirect (configured in Caddy) and, well, permanent redirects are called permanent for a reason (browsers cache them for unlimited time)… I’m afraid that the only solution for backward compatibility is to enable HTTPS with self-signed cert and override these redirects with another redirects… Simple workaround is to flush cache, open it in ~~porno~~ private mode, or open directly `http://<address>/dashboard`.

Comment (muhlpachr) [inline Dockerfile.armv7l]: 301 is not issue for .local and IP address URL (checked in Chrome, Firefox, MSIE) so we can safely fix to 302 without intemediate step.
Cache clear requirement for upgrade should not be considered as obstacle for aplha sw anyway, so why do it complicated ?

**Comment by jirutka** [comment]
Okay, I’ve rebased it. HEAD of the tree before rebase is 2ece1b00e8e094c5c070672fdc4f35ee601e2384.

**Pull request merged**